File: ir/attributeexpr_ir.py
from .expr_ir import ExprIR
from common.span import SourceSpan


class AttributeExprIR(ExprIR):
    def __init__(self, base: str, attr: str, span: SourceSpan):
        super().__init__(span=span, value=base)
        self.base = base
        self.attr = attr
        self.span = span

File: rust/src/ir/expr/attributeexpr_ir.rs
use crate::ir::{expr_ir::ExprIR, span_ir::SourceSpan};

#[derive(Debug, Clone)]
pub struct AttributeExprIR {
    pub target: Box<ExprIR>,
    pub attr: String,
    pub span: SourceSpan,
}

impl AttributeExprIR {
    pub fn new(target: ExprIR, attr: impl Into<String>, span: SourceSpan) -> Self {
        Self {
            target: Box::new(target),
            attr: attr.into(),
            span,
        }
    }
}
